.github/workflows/changelog.yml

```diff
@@ -0,0 +1,155 @@
+# This workflow makes sure contributors don't forget to add a changelog entry or explicitly opt-out of it.
+
+name: Changelog
+
+on:
+  pull_request:
+    types:
+      - opened
+      - ready_for_review
+      - reopened
+      - synchronize
+      - labeled
+      - unlabeled
+
+# This workflow runs for not-yet-reviewed external contributions and so it
+# intentionally has no write access and only limited read access to the
+# repository.
+permissions:
+  contents: read
+  pull-requests: write
+
+jobs:
+  check-changelog-entry:
+    name: "Check Changelog Entry"
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: "Changed files"
+        uses: dorny/paths-filter@de90cc6fb38fc0963ad72b210f1f284cd68cea36 # v3.0.2
+        id: changelog
+        with:
+          filters: |
+            unreleased:
+              - '.changes/unreleased/*.yaml'
+            backported:
+              - '.changes/backported/*.yaml'
+            changelog:
+              - 'CHANGELOG.md'
+            version:
+              - 'version/VERSION'
+
+      - name: "Check for changelog entry"
+        uses: actions/github-script@60a0d83039c74a4aee543508d2ffcb1c3799cdea # v7.0.1
+        with:
+          script: |
+            async function createOrUpdateChangelogComment(commentDetails, deleteComment) {
+                const commentStart = "## Changelog Warning"
+                
+                const body = commentStart + "\n\n" + commentDetails;
+                const { number: issue_number } = context.issue;
+                const { owner, repo } = context.repo;
+                
+                // List all comments
+                const allComments = (await github.rest.issues.listComments({
+                    issue_number,
+                    owner,
+                    repo,
+                })).data;
+
+                const existingComment = allComments.find(c => c.body.startsWith(commentStart));
```

Comment:
if you wanted to be double sure, the response should also contain the user who made the comment. You could check that is the same user we're executing as now. We can always come back and add this if we find people do start writing comments that start with `## Changelog Warning`

Reply:
Yeah, let's fix this if the need arises 👍 We will probably need to touch this script at some point again when new needs surface :)